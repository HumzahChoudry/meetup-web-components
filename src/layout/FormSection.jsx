// @flow
import React from 'react';
import cx from 'classnames';

import Card from './Card';
import Bounds from './Bounds';
import Section from './Section';

/**
 * Wrapping component to standardize the structure of form sections.
 * @module FormSection
 * @param {object} props React props
 * @return {React.element} Form section wrapping component
 */

type Props = {
	/** Whether the inner `<Section />` has a separator (extra spacing and a border on the bottom) */
	withSeparator?: boolean,

	/** Whether the component is in a loading state */
	isLoading?: boolean,

	/** Props to pass to the `<Loading />` component */
	loadingProps?: {
		color?: string,
		scrimColor?: string,
		size?: MediaSizes,
	},
	/** The child elements of the component */
	children: React$Node,

	/** Nearest DOM element's class name */
	className?: string,
};
function FormSection({
	className,
	children,
	withSeparator,
	loadingProps = {}, // eslint-disable-line no-unused-vars
	isLoading,
}: Props) {
	return (
		<Section hasSeparator={withSeparator} className={cx(className, 'border--none')}>
			<Card
				initialHeight
				flushUntil="large"
				isLoading={isLoading}
				loadingProps={loadingProps}
			>
				<Bounds narrow>
					<Section hasSeparator className="border--none padding--top">
						{children}
					</Section>
				</Bounds>
			</Card>
		</Section>
	);
}

export default FormSection;
