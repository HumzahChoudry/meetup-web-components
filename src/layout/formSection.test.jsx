import React from 'react';
import { shallow } from 'enzyme';

import FormSection from './FormSection';

describe('FormSection', () => {
	it('exists and contains form section wrapping elements with proper attributes', () => {
		expect(shallow(<FormSection />)).toMatchSnapshot();
	});
});
