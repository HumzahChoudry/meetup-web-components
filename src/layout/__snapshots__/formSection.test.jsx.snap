// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSection exists and contains form section wrapping elements with proper attributes 1`] = `
<WithLoading(Section)
  className="border--none"
>
  <WithLoading(Card)
    flushUntil="large"
    initialHeight={true}
    loadingProps={Object {}}
  >
    <WithLoading(Bounds)
      narrow={true}
    >
      <WithLoading(Section)
        className="border--none padding--top"
        hasSeparator={true}
      />
    </WithLoading(Bounds)>
  </WithLoading(Card)>
</WithLoading(Section)>
`;
